src/theory/fp/fp_converter.cpp

```diff
@@ -820,868 +820,469 @@ FpConverter::uf FpConverter::buildComponents(TNode current)
-                d_rmMap.insert(current, traits::RTZ());
-                break;
-              default: Unreachable() << "Unknown rounding mode"; break;
-            }
-          }
-          else
-          {
-            /******** Variables ********/
-            rm tmp(nm->mkNode(kind::ROUNDINGMODE_BITBLAST, current));
-            d_rmMap.insert(current, tmp);
-            d_additionalAssertions.push_back(tmp.valid());
+            case RoundingMode::ROUND_NEAREST_TIES_TO_EVEN:
+              d_rmMap.insert(cur, traits::RNE());
+              break;
+            case RoundingMode::ROUND_NEAREST_TIES_TO_AWAY:
+              d_rmMap.insert(cur, traits::RNA());
+              break;
+            case RoundingMode::ROUND_TOWARD_POSITIVE:
+              d_rmMap.insert(cur, traits::RTP());
+              break;
+            case RoundingMode::ROUND_TOWARD_NEGATIVE:
+              d_rmMap.insert(cur, traits::RTN());
+              break;
+            case RoundingMode::ROUND_TOWARD_ZERO:
+              d_rmMap.insert(cur, traits::RTZ());
+              break;
+            default: Unreachable() << "Unknown rounding mode"; break;
           }
         }
         else
         {
-          Unreachable() << "Unknown kind of type RoundingMode";
+          rm tmp(nm->mkNode(kind::ROUNDINGMODE_BITBLAST, cur));
+          d_rmMap.insert(cur, tmp);
+          d_additionalAssertions.push_back(tmp.valid());
         }
       }
-      // Returns a rounding-mode type so don't alter the return value
-    }
-    else if (t.isFloatingPoint())
-    {
-      fpMap::const_iterator i(d_fpMap.find(current));
-
-      if (i == d_fpMap.end())
+      else if (t.isFloatingPoint())
       {
-        if (Theory::isLeafOf(current, THEORY_FP))
+        /* ---- FloatingPoint constants and variables ------------- */
+        if (Theory::isLeafOf(cur, THEORY_FP))
         {
-          if (current.getKind() == kind::CONST_FLOATINGPOINT)
+          if (kind == kind::CONST_FLOATINGPOINT)
           {
-            /******** Constants ********/
             d_fpMap.insert(
-                current,
-                symfpu::unpackedFloat<traits>(current.getConst<FloatingPoint>()
-                                                  .getLiteral()
-                                                  ->getSymUF()));
+                cur,
+                symfpu::unpackedFloat<traits>(
+                    cur.getConst<FloatingPoint>().getLiteral()->getSymUF()));
           }
           else
           {
-            /******** Variables ********/
-            d_fpMap.insert(current, buildComponents(current));
+            d_fpMap.insert(cur, buildComponents(cur));
           }
         }
         else
         {
-          switch (current.getKind())
+          /* ---- FloatingPoint operators --------------------------- */
+          Assert(kind != kind::CONST_FLOATINGPOINT && kind != kind::VARIABLE
+                 && kind != kind::BOUND_VARIABLE && kind != kind::SKOLEM);
```

Comment:
I usually prefer to split `Assert(cond1 && cond2)` into `Assert(cond1); Assert(cond2);` since when the assertion fails you immediately know why.